src/stat.go

```diff
@@ -125,27 +150,79 @@ func (g *Commands) stat(relToRootPath string, file *File, depth int) error {
 	}
 
 	// Arbitrary value for throttle pause duration
+	// TODO Is this really needed now that everything is serialized?
+
 	throttle := time.Tick(1e9 / 5)
 
-	prettyFileStat(g.log.Logf, relToRootPath, file)
-	perms, permErr := g.rem.listPermissions(file.Id)
-	if permErr != nil {
-		return permErr
-	}
+	if !g.opts.Md5sum {
+		prettyFileStat(g.log.Logf, relToRootPath, file)
+		perms, permErr := g.rem.listPermissions(file.Id)
+		if permErr != nil {
+			return permErr
+		}
 
-	for _, perm := range perms {
-		prettyPermission(g.log.Logf, perm)
+		for _, perm := range perms {
+			prettyPermission(g.log.Logf, perm)
+		}
+	} else if file.Md5Checksum != "" {
+		g.log.Logf("%32s  %s\n", file.Md5Checksum, strings.TrimPrefix(relToRootPath, "/"))
 	}
 
-	if !file.IsDir {
-		return nil
-	}
+	if file.IsDir {
+		//remoteChildren := FileArray{}
+		var remoteChildren FileArray
+
+		for child := range g.rem.FindByParentId(file.Id, g.opts.Hidden) {
+			remoteChildren = append(remoteChildren, child)
+			<-throttle
+		}
 
-	remoteChildren := g.rem.FindByParentId(file.Id, g.opts.Hidden)
-	for child := range remoteChildren {
-		g.stat(filepath.Clean(relToRootPath+"/"+child.Name), child, depth)
-		<-throttle
+		if g.opts.Md5sum {
+			// TODO use g.sort instead of sort.stable
+			// i.e g.sort(remoteChildren,"name")
```

Comment:
you can add this code here to sort and create a specialized function for name comparisions e.g in https://github.com/odeke-em/drive/blob/master/src/sort.go#L208-209

```diff 
// src/sort.go 
        case AttrName:
-               return nilCmpOrProceed(func(l, r *File) bool { return l.Name < r.Name })
+               return nilCmpOrProceed(_stableNameLessCmp)
```